cluster/sbin/collect-debug-artifacts.sh

```diff
@@ -186,18 +225,51 @@ function collect_on_remote {
         # also check for the pid line and get the pid
         proc_id=`sed -n 's/.*Process ID: \([0-9]\+\)$/\1/p' ${l}`
         if [ "${verbose}" = "1" ]; then
-          echo "Adding file ${l} to the array"
+          echo "Adding latest copyright header file ${l} to the array"
         fi
         files+=($l)
         last_restart_log="$l"
+      else
+        if [ "${verbose}" = "1" ]; then
+          echo "Adding file ${l} to the array"
+        fi
+        files+=($l)
       fi
     done
 
     # get all the gfs files as well
-    for l in $( ls *.gfs 2> /dev/null )
+    for l in $( ls *.gfs* 2> /dev/null )
     do
       files+=($l)
     done
+  else 
+    if [ "${verbose}" = "1" ]; then
+      echo "collecting files based on modified time"
+    fi
+    files=()
+    for l in $( ls -t *.log* 2>/dev/null )
+    do
+      file_mod_epoch=`date +%s --date "$(ls  -l --time-style=long-iso $l | cut -d' ' -f6-7)"`
+      if [ "${file_mod_epoch}" -ge "${start_epoch}" -a "${file_mod_epoch}" -le "${end_epoch}" ]; then
+        if [ "${verbose}" = "1" ]; then
+          echo "${l} MOD TIME = ${file_mod_epoch}"
+          echo "Adding file ${l} to the array"
+        fi
+        files+=($l)
+      fi
+    done
+
+    for l in $( ls -t *.gfs* 2>/dev/null )
+    do
+      file_mod_epoch=`date +%s --date "$(ls  -l --time-style=long-iso $l | cut -d' ' -f6-7)"`
+      if [ "${file_mod_epoch}" -ge "${start_epoch}" -a "${file_mod_epoch}" -le "${end_epoch}" ]; then
```

Comment:
This condition may be bit trickier. What we really want is that given a time-range `start-end`, the `start` should also consider the log which contains that `start` (and likewise for `end`). This means that if a log was created on `start-400` and continued till `start+400` then it should be included and similar for `end`.

There is a creation timestamp available in stat (%W) which could have been used but unfortunately most filesystems do not record this and it will give output as zero. Hence to accomplish this need to consider the modification timestamp of the previous file in the same directory and check if `start` lies between previous file's mod-time and this file's, and include this file if that is the case (likewise for `end`).